pkg/virt-controller/services/template.go

```diff
@@ -1760,6 +1760,16 @@ func addProbeOverhead(probe *v1.Probe, to *resource.Quantity) bool {
 }
 
 func updateProbe(vmi *v1.VirtualMachineInstance, computeProbe *k8sv1.Probe) {
+	if vmi.Spec.ReadinessProbe == nil && vmi.Spec.LivenessProbe == nil {
+		return
+	}
+	if vmi.Spec.ReadinessProbe != nil {
```

Comment:
This can also be removed I think. We want to do this for both Readiness and Liveness, right?
I agree that a guest-ping on a livenessProbe will probably be rare, but I'd prefer not to differentiate between the two in what functionality we provide.

Reply:
- Cannot remove this check because Readiness probe can be nil when we're just calling this func for liveness, or other way
- Other way is to just have separate `updateProbe` funcs for liveness and readiness probes, but that's equally bad readability 
- Added the check for liveness probe, agree on having an option for guest-ping with it

Reply:
I still wish we could make this a little cleaner.
Maybe moving the if statements together `if vmi.Spec.LivenessProbe != nil && vmi.Spec.LivenessProbe.GuestAgentPing != nil {` would help?
Or refactoring the above part where updateProbe gets used a little more :thinking:

Reply:
But I'm fine with it if you are.

Reply:
`if vmi.Spec.LivenessProbe != nil && vmi.Spec.LivenessProbe.GuestAgentPing != nil ` would still cause nil pointer error when only Readiness probe is defined

Reply:
Just split the func into separate ones for readiness and liveness, I think it makes it more readable